cluster-agent/utils/argocd_login_command_test.go

```diff
@@ -29,4 +34,118 @@ var _ = Describe("ArgoCD Login Command", func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 	})
+
+	Context("Test generateDefaultClientForServerAddress", func() {
+		var (
+			server = "http://argocd-server"
+			token  = "sample-token-ds32n"
+		)
+		It("should return a valid argocd client when skipTLS is enabled", func() {
+			skipTLS := true
+			client, err := generateDefaultClientForServerAddress(server, token, tlsConfig{skipTLSTest: skipTLS})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(client).ToNot(BeNil())
+
+			expectedClientOptions := apiclient.ClientOptions{
+				ServerAddr:  server,
+				AuthToken:   token,
+				Insecure:    true,
+				PlainText:   false,
+				GRPCWeb:     false,
+				PortForward: false,
+			}
+			Expect(client.ClientOptions()).To(Equal(expectedClientOptions))
+		})
+
+		It("should return a valid argocd client when skipTLS is disabled", func() {
+			skipTLS := false
+			mockTestTLS := func(addr string, dialtime time.Duration) (*grpc.TLSTestResult, error) {
+				tlsResult := &grpc.TLSTestResult{
+					TLS: true,
+				}
+				return tlsResult, nil
+			}
+
+			client, err := generateDefaultClientForServerAddress(server, token, tlsConfig{
+				skipTLSTest: skipTLS,
+				testTLS:     mockTestTLS,
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(client).ToNot(BeNil())
+
+			expectedClientOptions := apiclient.ClientOptions{
+				ServerAddr:  server,
+				AuthToken:   token,
+				Insecure:    true,
+				PlainText:   false,
+				GRPCWeb:     false,
+				PortForward: false,
+			}
+			Expect(client.ClientOptions()).To(Equal(expectedClientOptions))
+		})
+
+		It("should return an error when skipTLS is false and the TLS verification fails", func() {
+			skipTLS := false
+			expectedErr := fmt.Errorf("random TLS error")
+			mockTestTLS := func(addr string, dialtime time.Duration) (*grpc.TLSTestResult, error) {
+				return nil, expectedErr
+			}
+
+			client, err := generateDefaultClientForServerAddress(server, token, tlsConfig{
+				skipTLSTest: skipTLS,
+				testTLS:     mockTestTLS,
+			})
+
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(Equal(expectedErr))
+			Expect(client).To(BeNil())
+		})
+
+		It("should return an error when skipTLS is false but the server doesn't support TLS", func() {
+			skipTLS := false
+			expectedErr := fmt.Errorf("server is not configured with TLS")
+			mockTestTLS := func(addr string, dialtime time.Duration) (*grpc.TLSTestResult, error) {
+				return &grpc.TLSTestResult{
+					TLS: false,
+				}, nil
+			}
+
+			client, err := generateDefaultClientForServerAddress(server, token, tlsConfig{
+				skipTLSTest: skipTLS,
+				testTLS:     mockTestTLS,
+			})
+
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(Equal(expectedErr))
+			Expect(client).To(BeNil())
+		})
+
+		It("should not return an error when insecure is true and the server certificate had errors", func() {
```

Comment:
I think it should ->  should not return an error when insecure `field` is true, which will be more readable, WDYT?